plugins/com.google.cloud.tools.eclipse.appengine.ui/src/com/google/cloud/tools/eclipse/appengine/ui/AppEngineLibrariesSelectorGroup.java

```diff
@@ -107,55 +104,209 @@ private static String getLibraryName(Library library) {
-
-        WritableValue intermediate = new WritableValue(false, Boolean.class);
-        bindingContext.bindValue(libraryButtonSelection, intermediate);
-        bindingContext.bindValue(dependencyButtonSelection, intermediate);
-        bindingContext.bindValue(dependencyButtonEnablement, intermediate,
-                                 new UpdateValueStrategy().setConverter(BooleanConverter.negate()),
-                                 new UpdateValueStrategy().setConverter(BooleanConverter.negate()));
-      }
+  private void addDatabindingForButton(final Button libraryButton) {
+    final Library library = (Library) libraryButton.getData();
+    ISWTObservableValue libraryButtonSelection = WidgetProperties.selection().observe(libraryButton);
+    ISWTObservableValue libraryButtonEnablement = WidgetProperties.enabled().observe(libraryButton);
+    // library selection UI -> model
+    bindingContext.bindValue(libraryButtonSelection,
+                             new NullComputedValue(getDisplayRealm()),
+                             new UpdateValueStrategy().setConverter(new HandleLibrarySelectionConverter(selectedLibraries,
+                                                                                                        library)),
+                             new UpdateValueStrategy(UpdateValueStrategy.POLICY_NEVER));
+    // library selection model -> UI
+    bindingContext.bindValue(libraryButtonSelection, 
+                             new DependentLibrarySelected(getDisplayRealm(),
+                                                          selectedLibraries,
+                                                          library.getId(),
+                                                          true /* resultIfFound */,
+                                                          new ButtonManuallySelected(libraryButton)),
+                             new UpdateValueStrategy(UpdateValueStrategy.POLICY_NEVER),
+                             new UpdateValueStrategy());
+    // library checkbox enablement model -> UI
+    bindingContext.bindValue(libraryButtonEnablement,
+                             new DependentLibrarySelected(getDisplayRealm(),
+                                                          selectedLibraries,
+                                                          library.getId(),
+                                                          false /* resultIfFound */),
+                             new UpdateValueStrategy(UpdateValueStrategy.POLICY_NEVER),
+                             new UpdateValueStrategy());
+  }
+
+  public void dispose() {
+    if (bindingContext != null) {
+      bindingContext.dispose();
+    }
+  }
+
+  private Realm getDisplayRealm() {
+    return DisplayRealm.getRealm(parentContainer.getDisplay());
+  }
+
+  @VisibleForTesting
+  List<Button> getLibraryButtons() {
+    return libraryButtons;
+  }
+
+  /**
+   * Utility interface to return a value;
+   */
+  private static interface Getter<T> {
```

Comment:
I was thinking about it, but that declares an `Exception` thrown in the signature and I didn't like that. I was hoping there's a simple interface I forgot about instead of this.